File: Algorithms/Python3.x/485-Max_Consecutive_Ones.py
# Runtime: 380 ms, faster than 98.79% of Python3 online submissions for Max Consecutive Ones.
# Memory Usage: 12.9 MB, less than 100.00% of Python3 online submissions for Max Consecutive Ones.
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = 0
        tmp = 0
        for i in range(len(nums)):
            if nums[i] == 1:
                tmp += 1
            else:
                ans = max(ans, tmp)
                tmp = 0
        return max(ans, tmp)

File: Algorithms/C/485-Max_Consecutive_Ones.c
// Runtime: 40 ms, faster than 63.16% of C online submissions for Max Consecutive Ones.
// Memory Usage: 8.6 MB, less than 100.00% of C online submissions for Max Consecutive Ones.
int findMaxConsecutiveOnes(int* nums, int numsSize){
    int ans = 0;
    int tmp = 0;
    for (int i = 0; i < numsSize; ++i)
        if (*(nums + i) == 1)
            tmp++;
        else {
            ans = ans > tmp ? ans : tmp;
            tmp = 0;
        }
    return ans > tmp ? ans : tmp;
}
